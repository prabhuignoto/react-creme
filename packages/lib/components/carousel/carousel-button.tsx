import classNames from 'classnames';
import React from 'react';
import { ChevronRightIcon } from '../../icons';
import { Button } from '../button/button';
import { CarouselButtonProps } from './carousel-model';
import './carousel.scss';

const CarouselButton: React.FunctionComponent<CarouselButtonProps> = ({
  onClick,
  position,
  hide,
  direction,
  focusable,
  label,
}: CarouselButtonProps) => {
  return (
    <span
      className={classNames(
        [
          'rc-carousel-btn',
          `rc-carousel-btn-${position}`,
          `rc-carousel-btn-${direction}`,
        ],
        {
          'rc-carousel-btn-hide': hide,
        }
      )}
    >
      <Button type="icon" onClick={onClick} focusable={focusable} label={label}>
        <ChevronRightIcon />
      </Button>
    </span>
  );
};

CarouselButton.displayName = 'CarouselButton';

export { CarouselButton };
