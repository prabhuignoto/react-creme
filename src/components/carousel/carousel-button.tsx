import classNames from "classnames";
import React from "react";
import { Button } from "..";
import { ChevronRightIcon } from "../../icons";
import "./carousel.scss";

interface CarouselButtonModel {
  onClick: () => void;
  position: "left" | "right";
}

const CarouselButton: React.FunctionComponent<CarouselButtonModel> = React.memo(
  ({ onClick, position }: CarouselButtonModel) => {
    return (
      <span
        className={classNames([
          "rc-carousel-btn",
          `rc-carousel-btn-${position}`,
        ])}
      >
        <Button type="icon" onClick={onClick}>
          <ChevronRightIcon />
        </Button>
      </span>
    );
  }
);

CarouselButton.displayName = "CarouselButton";

export { CarouselButton };
