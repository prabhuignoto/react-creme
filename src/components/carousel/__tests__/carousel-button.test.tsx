import { fireEvent, render } from "@testing-library/react";
import React from "react";
import { CarouselButton } from "../carousel-button";

const handler = jest.fn();

describe("Carousel Button", () => {
  it("should render with position", () => {
    const { container } = render(
      <CarouselButton position="left" onClick={handler} />
    );

    expect(container.firstChild).toBeInTheDocument();
    expect(container.firstChild).toHaveClass("rc-carousel-btn-left");
  });

  it("should call the onClick handler", () => {
    const { container } = render(
      <CarouselButton position="left" onClick={handler} />
    );

    if (container.firstChild) {
      fireEvent.click(container.firstChild);

      expect(handler).toBeCalled();
    }
  });
});
