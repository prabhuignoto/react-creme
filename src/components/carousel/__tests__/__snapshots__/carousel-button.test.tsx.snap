// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Carousel Button should render carousel button snapshot 1`] = `
<button
  class="rc-btn-sm rc-btn-icon rc-btn"
  role="button"
  tabindex="0"
>
  <span
    class="rc-btn-icon-container"
  >
    <svg
      fill="none"
      height="24"
      stroke="currentColor"
      stroke-linecap="round"
      stroke-linejoin="round"
      stroke-width="2"
      viewBox="0 0 24 24"
      width="24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M9 18l6-6-6-6"
      />
    </svg>
  </span>
</button>
`;
